dict/dict0dict.cc

```diff
@@ -3885,7 +3819,208 @@ void dict_close() {
   mem_free(dict_sys);
   dict_sys = nullptr;
 
-  for (ulint i = 0; i < DICT_INDEX_STAT_MUTEX_SIZE; i++) {
-    mutex_free(&dict_index_stat_mutex[i]);
+  for (auto &i : dict_index_stat_mutex) {
+    mutex_free(&i);
   }
 }
+
+dict_table_lru::dict_table_lru(std::int32_t capacity, std::function<dict_table_t *(std::uint64_t)> table_fetcher)
+    : m_capacity{capacity}, m_size{0}, m_table_fetcher{std::move(table_fetcher)}, m_tables{}, m_name_lookup{}, m_id_lookup{} {}
+
+dict_table_t *dict_table_lru::get(std::string_view name) {
+  if (auto itr = m_name_lookup.find(name); itr != m_name_lookup.end()) {
+    ut_a(m_name_lookup[name] == m_id_lookup[itr->second->id]);
+    move_to_front(itr->second);
+    return (*itr->second->list_itr);
+  }
+
+  return nullptr;
+}
+
+dict_table_t *dict_table_lru::get(std::uint64_t id) {
+  if (auto itr = m_id_lookup.find(id); itr != m_id_lookup.end()) {
+    ut_a(m_id_lookup[id] == m_name_lookup[itr->second->name]);
+    move_to_front(itr->second);
+    return (*itr->second->list_itr);
+  }
+
+  return nullptr;
+}
+
+bool dict_table_lru::contains(std::uint64_t id) {
+  ib_logger(ib_stream, "contains check table: %lu\n", id);
```

Comment:
When writing new error logging messages use log_err(), log_warn(), log_info() etc. I want to get rid of this ib_stream mechanism.